src/view/pages/statement/components/map/StatementMap.tsx

```diff
@@ -70,15 +73,42 @@ const StatementMap: FC<Props> = ({ statement }) => {
 
 		const topResult = sortStatementsByHirarrchy([
 			statement,
-			...childStatements.filter((state) => state.statementType !== "statement"),
+			...childStatements.filter((state) => isOptionFn(state) || state.statementType === StatementType.question),
 		])[0];
 
 		setResults(topResult);
 	};
 
+	const dispatch = useAppDispatch();
+
 	useEffect(() => {
-		getSubStatements();
-	}, []);
+		let unsubscribe: Unsubscribe | null = null;
+
+		const fetchInitialData = async () => {
+			try {
+				unsubscribe = await listenToChildStatements(dispatch, statement.statementId, (childStatements) => {
+					setSubStatements(childStatements);
+
+					const topResult = sortStatementsByHirarrchy([
+						statement,
+						...childStatements.filter((state) => isOptionFn(state) || state.statementType === StatementType.question),
+					])[0];
+
+					setResults(topResult);
```

Comment:
Create the callback function outside of the listenToChildStatements props to improve readability.